### ISSUE TITLE
Setting up environment variables

### ISSUE THREAD

**Issue by edgarcosta** [sagemathinc/cocalc#2855]
I would like to setup environment variables when working on a jupyter notebook, so I can load my local libraries.

I tried three workarounds, and didn't get anywhere:

1 - setting the variables in ~/.bashrc  Indeed, it seems every kernel ignores these with the exception of bash.

2- I tried using the magic `%env LD_LIBRARY_PATH = /home/user/local/lib` this did not seem to affect ipython in practice, as iPyhon still doesn't search for my local libraries.

3- I don't expect this one to work, but I still tried:
```
import os 
os.environ['LD_LIBRARY_PATH'] = '/home/user/local/lib'
```

Any thoughts?

ps: everything works fine if I run sage from the commandline

**Comment by shyams2** [comment]
Is there any progress / info on this? I'm trying to do something similar

**Comment by haraldschilly** [comment]
well, depends a bit on your case. for those where the library isn't loaded when the kernel starts, setting it via `os.environ[ ... ] = ...` should work. This needs to be done before a library is imported.

Regarding the kernel startup itself, IIRC there were slight changes to how it is processed. The basic idea is to copy the kernel you want to modify (into your own project) and then add the env variables to the configuration. Maybe I should try it with a python kernel and document it. @shyams2 which kernel do you want to work with?

**Comment by haraldschilly** [comment]
Hi, I tested this.

1. I created a custom kernel according to https://doc.cocalc.com/howto/custom-jupyter-kernel.html by copying the global `python3` kernel. In my case, I named it `python3_custom`.
2. then I opened $HOME/.local/share/jupyter/kernels/python3_custom/kernel.json` and edited the display name and added the `env` key like
```
  "env": {
    "MyCustomVar": "foobar"
  },
```
3. Finally, after reloading the kernel list from the kernel menu in a jupyter notebook, I checked if that variable exists:

![Screenshot from 2019-06-28 17-43-37](https://user-images.githubusercontent.com/207405/60354329-45817180-99cc-11e9-89f5-6e1f18df1f56.png)
